File: IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgramVRFactory.m
//
//  IRGLProgramVRFactory.m
//  IRPlayer
//
//  Created by Phil on 2019/8/21.
//  Copyright © 2019 Phil. All rights reserved.
//

#import "IRGLProgramVRFactory.h"
#import <IRPlayer_swift/IRPlayer_swift-Swift.h>

@implementation IRGLProgramVRFactory

-(IRGLProgram2D*)createIRGLProgramWithPixelFormat:(IRPixelFormat)pixelFormat withViewprotRange:(CGRect)viewprotRange withParameter:(IRMediaParameter*)parameter{
    return [IRGLProgramFactory createIRGLProgramVRWithPixelFormat:pixelFormat viewportRange:viewprotRange parameter:(IRMediaParameter*)parameter];
}

@end

File: Sources/IRPlayer-swift/Class/Core/Display/RenderKit/Program/Factory/IRGLProgramVRFactory.swift
//
//  IRGLProgramVRFactory.swift
//  IRPlayer-swift
//
//  Created by irons on 2024/9/13.
//

import Foundation

class IRGLProgramVRFactory: IRGLProgram2DFactory {

    override func createIRGLProgram(pixelFormat: IRPixelFormat, viewportRange: CGRect, parameter: IRMediaParameter?) -> IRGLProgram2D {
        return IRGLProgramFactory.createIRGLProgramVR(pixelFormat: pixelFormat, viewportRange: viewportRange, parameter: parameter)
    }
}
